analysis/tools/ccsh/src/main/kotlin/de/maibornwolff/codecharta/tools/ccsh/Ccsh.kt

```diff
@@ -75,15 +87,80 @@ class Ccsh : Callable<Void?> {
         fun executeCommandLine(args: Array<String>): Int {
             val commandLine = CommandLine(Ccsh())
             commandLine.executionStrategy = CommandLine.RunAll()
-            if (args.isEmpty() || isParserUnknown(args, commandLine)) {
-                return executeInteractiveParser(commandLine)
+            return if (args.isEmpty()) {
+                val configuredParsers = offerInteractiveParserSuggestions(commandLine)
+                if (configuredParsers.isEmpty()) {
+                    return 0
+                }
+
+                return executeConfiguredParsers(commandLine, configuredParsers)
+            } else if (isParserUnknown(args, commandLine) || args.contains("--interactive") || args.contains("-i")) {
+                executeInteractiveParser(commandLine)
             } else {
-                return commandLine.execute(*sanitizeArgs(args))
+                commandLine.execute(*sanitizeArgs(args))
             }
         }
 
+        fun offerInteractiveParserSuggestions(commandLine: CommandLine): Map<String, List<String>> {
```

Comment:
This method is more like `get...` method.
You could also think about to extract this functionality to InteractiveParserSuggestionDialog class.

Reply:
I refactored it a bit: Now it only returns all applicable parsers but does not use the Kinquirer prompt. This is done in the calling function (ccsh.kt) now since all other prompts are also created there.